app/components/donation/credit-card.js

```diff
@@ -2,11 +2,89 @@ import Ember from 'ember';
 
 const {
   Component,
-  computed: { not }
+  computed,
+  computed: { and, not, or },
+  inject: { service }
 } = Ember;
 
 export default Component.extend({
   classNames: ['credit-card-form'],
-  canDonate: true,
-  cannotDonate: not('canDonate')
+  month: '',
+  months: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
+  year: '',
+  years: [],
+
+  /**
+    @property stripe
+    @type Ember.Service
+   */
+  stripe: service(),
+
+  cardInvalid: not('cardValid'),
+  cardValid: and('isCardNumberValid', 'isCVCValid', 'isExpiryValid'),
+
+  preventSubmit: or('isSubmitting', 'cardInvalid'),
+
+  isSubmitting: false,
+
+  date: computed('month', 'year', function() {
```

Comment:
Thoughts on simply renaming this to `dateString` so it is clear we expect it to be a string and not a date object?